routing/router.go

```diff
@@ -817,25 +817,48 @@ func (r *ChannelRouter) pruneZombieChans() error {
 			info.ChannelID)
 
 		// TODO(roasbeef): add ability to delete single directional edge
-		chansToPrune = append(chansToPrune, info.ChannelID)
+		chansToPrune[info.ChannelID] = true
```

Comment:
can this map also be used to exit this function early if the channel has already been processed?

Reply:
Good one. added logic at the beginning of the function.